// Code generated by mockery v2.53.3. DO NOT EDIT.

package mocks

import (
	context "context"

	domain "github.com/assimoes/beautix/internal/domain"
	mock "github.com/stretchr/testify/mock"

	time "time"

	uuid "github.com/google/uuid"
)

// AvailabilityExceptionService is an autogenerated mock type for the AvailabilityExceptionService type
type AvailabilityExceptionService struct {
	mock.Mock
}

// CountBusinessAvailabilityExceptions provides a mock function with given fields: ctx, businessID
func (_m *AvailabilityExceptionService) CountBusinessAvailabilityExceptions(ctx context.Context, businessID uuid.UUID) (int64, error) {
	ret := _m.Called(ctx, businessID)

	if len(ret) == 0 {
		panic("no return value specified for CountBusinessAvailabilityExceptions")
	}

	var r0 int64
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID) (int64, error)); ok {
		return rf(ctx, businessID)
	}
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID) int64); ok {
		r0 = rf(ctx, businessID)
	} else {
		r0 = ret.Get(0).(int64)
	}

	if rf, ok := ret.Get(1).(func(context.Context, uuid.UUID) error); ok {
		r1 = rf(ctx, businessID)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// CreateAvailabilityException provides a mock function with given fields: ctx, input
func (_m *AvailabilityExceptionService) CreateAvailabilityException(ctx context.Context, input *domain.CreateAvailabilityExceptionInput) (*domain.AvailabilityException, error) {
	ret := _m.Called(ctx, input)

	if len(ret) == 0 {
		panic("no return value specified for CreateAvailabilityException")
	}

	var r0 *domain.AvailabilityException
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, *domain.CreateAvailabilityExceptionInput) (*domain.AvailabilityException, error)); ok {
		return rf(ctx, input)
	}
	if rf, ok := ret.Get(0).(func(context.Context, *domain.CreateAvailabilityExceptionInput) *domain.AvailabilityException); ok {
		r0 = rf(ctx, input)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*domain.AvailabilityException)
		}
	}

	if rf, ok := ret.Get(1).(func(context.Context, *domain.CreateAvailabilityExceptionInput) error); ok {
		r1 = rf(ctx, input)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// DeleteAvailabilityException provides a mock function with given fields: ctx, id, deletedBy
func (_m *AvailabilityExceptionService) DeleteAvailabilityException(ctx context.Context, id uuid.UUID, deletedBy uuid.UUID) error {
	ret := _m.Called(ctx, id, deletedBy)

	if len(ret) == 0 {
		panic("no return value specified for DeleteAvailabilityException")
	}

	var r0 error
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID, uuid.UUID) error); ok {
		r0 = rf(ctx, id, deletedBy)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// GetAvailabilityException provides a mock function with given fields: ctx, id
func (_m *AvailabilityExceptionService) GetAvailabilityException(ctx context.Context, id uuid.UUID) (*domain.AvailabilityException, error) {
	ret := _m.Called(ctx, id)

	if len(ret) == 0 {
		panic("no return value specified for GetAvailabilityException")
	}

	var r0 *domain.AvailabilityException
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID) (*domain.AvailabilityException, error)); ok {
		return rf(ctx, id)
	}
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID) *domain.AvailabilityException); ok {
		r0 = rf(ctx, id)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*domain.AvailabilityException)
		}
	}

	if rf, ok := ret.Get(1).(func(context.Context, uuid.UUID) error); ok {
		r1 = rf(ctx, id)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// GetStaffAvailabilityExceptions provides a mock function with given fields: ctx, staffID
func (_m *AvailabilityExceptionService) GetStaffAvailabilityExceptions(ctx context.Context, staffID uuid.UUID) ([]*domain.AvailabilityException, error) {
	ret := _m.Called(ctx, staffID)

	if len(ret) == 0 {
		panic("no return value specified for GetStaffAvailabilityExceptions")
	}

	var r0 []*domain.AvailabilityException
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID) ([]*domain.AvailabilityException, error)); ok {
		return rf(ctx, staffID)
	}
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID) []*domain.AvailabilityException); ok {
		r0 = rf(ctx, staffID)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).([]*domain.AvailabilityException)
		}
	}

	if rf, ok := ret.Get(1).(func(context.Context, uuid.UUID) error); ok {
		r1 = rf(ctx, staffID)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// GetStaffAvailabilityExceptionsByDateRange provides a mock function with given fields: ctx, staffID, start, end
func (_m *AvailabilityExceptionService) GetStaffAvailabilityExceptionsByDateRange(ctx context.Context, staffID uuid.UUID, start time.Time, end time.Time) ([]*domain.AvailabilityException, error) {
	ret := _m.Called(ctx, staffID, start, end)

	if len(ret) == 0 {
		panic("no return value specified for GetStaffAvailabilityExceptionsByDateRange")
	}

	var r0 []*domain.AvailabilityException
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID, time.Time, time.Time) ([]*domain.AvailabilityException, error)); ok {
		return rf(ctx, staffID, start, end)
	}
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID, time.Time, time.Time) []*domain.AvailabilityException); ok {
		r0 = rf(ctx, staffID, start, end)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).([]*domain.AvailabilityException)
		}
	}

	if rf, ok := ret.Get(1).(func(context.Context, uuid.UUID, time.Time, time.Time) error); ok {
		r1 = rf(ctx, staffID, start, end)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// ListBusinessAvailabilityExceptions provides a mock function with given fields: ctx, businessID, page, pageSize
func (_m *AvailabilityExceptionService) ListBusinessAvailabilityExceptions(ctx context.Context, businessID uuid.UUID, page int, pageSize int) ([]*domain.AvailabilityException, error) {
	ret := _m.Called(ctx, businessID, page, pageSize)

	if len(ret) == 0 {
		panic("no return value specified for ListBusinessAvailabilityExceptions")
	}

	var r0 []*domain.AvailabilityException
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID, int, int) ([]*domain.AvailabilityException, error)); ok {
		return rf(ctx, businessID, page, pageSize)
	}
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID, int, int) []*domain.AvailabilityException); ok {
		r0 = rf(ctx, businessID, page, pageSize)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).([]*domain.AvailabilityException)
		}
	}

	if rf, ok := ret.Get(1).(func(context.Context, uuid.UUID, int, int) error); ok {
		r1 = rf(ctx, businessID, page, pageSize)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// ListBusinessAvailabilityExceptionsByDateRange provides a mock function with given fields: ctx, businessID, start, end, page, pageSize
func (_m *AvailabilityExceptionService) ListBusinessAvailabilityExceptionsByDateRange(ctx context.Context, businessID uuid.UUID, start time.Time, end time.Time, page int, pageSize int) ([]*domain.AvailabilityException, error) {
	ret := _m.Called(ctx, businessID, start, end, page, pageSize)

	if len(ret) == 0 {
		panic("no return value specified for ListBusinessAvailabilityExceptionsByDateRange")
	}

	var r0 []*domain.AvailabilityException
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID, time.Time, time.Time, int, int) ([]*domain.AvailabilityException, error)); ok {
		return rf(ctx, businessID, start, end, page, pageSize)
	}
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID, time.Time, time.Time, int, int) []*domain.AvailabilityException); ok {
		r0 = rf(ctx, businessID, start, end, page, pageSize)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).([]*domain.AvailabilityException)
		}
	}

	if rf, ok := ret.Get(1).(func(context.Context, uuid.UUID, time.Time, time.Time, int, int) error); ok {
		r1 = rf(ctx, businessID, start, end, page, pageSize)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// UpdateAvailabilityException provides a mock function with given fields: ctx, id, input, updatedBy
func (_m *AvailabilityExceptionService) UpdateAvailabilityException(ctx context.Context, id uuid.UUID, input *domain.UpdateAvailabilityExceptionInput, updatedBy uuid.UUID) error {
	ret := _m.Called(ctx, id, input, updatedBy)

	if len(ret) == 0 {
		panic("no return value specified for UpdateAvailabilityException")
	}

	var r0 error
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID, *domain.UpdateAvailabilityExceptionInput, uuid.UUID) error); ok {
		r0 = rf(ctx, id, input, updatedBy)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// NewAvailabilityExceptionService creates a new instance of AvailabilityExceptionService. It also registers a testing interface on the mock and a cleanup function to assert the mocks expectations.
// The first argument is typically a *testing.T value.
func NewAvailabilityExceptionService(t interface {
	mock.TestingT
	Cleanup(func())
}) *AvailabilityExceptionService {
	mock := &AvailabilityExceptionService{}
	mock.Mock.Test(t)

	t.Cleanup(func() { mock.AssertExpectations(t) })

	return mock
}
